contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -5325,3 +5336,169 @@ pltsql_remove_current_query_env(void)
+										 &relation,
+										 AccessShareLock,
+										 true);
+
+			if (!relation)
+				return;
+
+			/* Get major_name */
+			major_name = pstrdup(RelationGetRelationName(relation));
+			relation_close(relation, AccessShareLock);
+
+			/* Get schema_name */
+			schema_oid = get_object_namespace(&address);
+			if (OidIsValid(schema_oid))
+			{
+				physical_schema_name = get_namespace_name(schema_oid);
+				if (physical_schema_name)
+				{
+					logical_schema_name =
+						get_logical_schema_name(physical_schema_name, true);
+					pfree(physical_schema_name);
+
+					if (logical_schema_name)
+					{
+						schema_name =
+							downcase_identifier(logical_schema_name,
+												strlen(logical_schema_name),
+												false, false);
+						pfree((char *) logical_schema_name);
+					}
+				}
+			}
+
+			if (schema_name && major_name)
+			{
+				delete_extended_property(3, db_id, schema_name, major_name,
+										 NULL, NULL);
+			}
+
+			if (schema_name)
+				pfree(schema_name);
+			if (major_name)
+				pfree(major_name);
+		}
+	}
+	else if (stmt->removeType == OBJECT_PROCEDURE ||
+			 stmt->removeType == OBJECT_FUNCTION ||
+			 stmt->removeType == OBJECT_TYPE)
+	{
+		HeapTuple	tuple;
+
+		foreach(cell, stmt->objects)
+		{
+			address = get_object_address(stmt->removeType,
+										 lfirst(cell),
+										 &relation,
+										 AccessShareLock,
+										 true);
+			Assert(relation == NULL);
+			if (!OidIsValid(address.objectId))
+				return;
+
+			/* Get major_name */
+			relation = table_open(address.classId, AccessShareLock);
+			tuple = get_catalog_object_by_oid(relation,
+											  get_object_attnum_oid(address.classId),
+											  address.objectId);
+			if (!HeapTupleIsValid(tuple))
+			{
+				table_close(relation, AccessShareLock);
+				return;
+			}
+
+			if (stmt->removeType == OBJECT_PROCEDURE ||
+				stmt->removeType == OBJECT_FUNCTION)
+				major_name = pstrdup(NameStr(((Form_pg_proc) GETSTRUCT(tuple))->proname));
+			else if (stmt->removeType == OBJECT_TYPE)
+				major_name = pstrdup(NameStr(((Form_pg_type) GETSTRUCT(tuple))->typname));
+
+			table_close(relation, AccessShareLock);
+
+			/* Get schema_name */
+			schema_oid = get_object_namespace(&address);
+			if (OidIsValid(schema_oid))
+			{
+				physical_schema_name = get_namespace_name(schema_oid);
+				if (physical_schema_name)
+				{
+					logical_schema_name =
+						get_logical_schema_name(physical_schema_name, true);
+					pfree(physical_schema_name);
+
+					if (logical_schema_name)
+					{
+						schema_name =
+							downcase_identifier(logical_schema_name,
+												strlen(logical_schema_name),
+												false, false);
+						pfree((char *) logical_schema_name);
+					}
+				}
+			}
+
+			if (schema_name && major_name)
+			{
+				delete_extended_property(3, db_id, schema_name, major_name,
+										 NULL, NULL);
+			}
+
+			if (schema_name)
+				pfree(schema_name);
+			if (major_name)
+				pfree(major_name);
+		}
+	}
```

Comment:
Do we have any other objects to handle ? If not, how about add an error handling case?